src/Diagnostics.RuntimeHost/Utilities/RequestBodyValidator.cs

```diff
@@ -20,5 +23,34 @@ public static bool ValidateRequestBody(JToken jsonBody, string[] fieldNames, out
             outputMessage = string.Concat(validationResults);
             return !(validationResults.Count() > 0);
         }
+
+        public static bool ValidateRequestBody(IRequestBodyBase body, out string outputMessage)
+        {
+            outputMessage = "";
+            if (body == null)
+            {
+                outputMessage = "Missing request body";
+                return false;
+            }
+
+            List<string> missingProperties = new List<string>();
+
+            foreach (var property in body.GetType().GetProperties())
+            {
+                if (property.GetValue(body) == null)
+                {
+                    missingProperties.Add(property.Name);
+                }
+            }
+
+            if (!(missingProperties.Count() > 0))
```

Comment:
nit: it's ok with: if (missingProperties.Count() == 0) because Count() never returns a negative value.